worker/dbaccessor/worker.go

```diff
@@ -539,15 +552,29 @@ func (w *dbWorker) initialiseDqlite(options ...app.Option) error {
 // Since GetDB blocks until dbReady is closed, and initialiseDqlite waits for
 // the node to be ready, we can assume that we will never race with a nil dbApp
 // when first starting up.
-// Since the only way we can get into this race is during shutdown, it is safe
-// to return ErrDying if we detect a nil database.
-// This preserves the error that the worker is exiting with, including nil.
+// Since the only way we can get into this race is during shutdown or a rebind.
+// It is safe to return ErrDying if the catacomb is dying when we detect a nil
```

Comment:
This could be more incisive, but I'm happy to land and do a follow-up edit myself.